File: module_user_center/src/main/java/com/wil/user/di/module/SettingsModule.java
package com.wil.user.di.module;

import com.jess.arms.di.scope.ActivityScope;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

import com.wil.user.mvp.contract.SettingsContract;
import com.wil.user.mvp.model.SettingsModel;


/**
 * ================================================
 * Description:
 * <p>
 * Created by MVPArmsTemplate on 08/15/2019 22:50
 * <a href="mailto:jess.yan.effort@gmail.com">Contact me</a>
 * <a href="https://github.com/JessYanCoding">Follow me</a>
 * <a href="https://github.com/JessYanCoding/MVPArms">Star me</a>
 * <a href="https://github.com/JessYanCoding/MVPArms/wiki">See me</a>
 * <a href="https://github.com/JessYanCoding/MVPArmsTemplate">模版请保持更新</a>
 * ================================================
 */
@Module
public abstract class SettingsModule {

    @Binds
    abstract SettingsContract.Model bindSettingsModel(SettingsModel model);
}

File: module_user_center/src/main/java/com/wil/user/di/module/SettingsModule.kt
package com.wil.user.di.module

import com.jess.arms.di.scope.ActivityScope

import dagger.Binds
import dagger.Module
import dagger.Provides

import com.wil.user.mvp.contract.SettingsContract
import com.wil.user.mvp.model.SettingsModel


/**
 * ================================================
 * Description:
 *
 *
 * Created by MVPArmsTemplate on 08/15/2019 22:50
 * [Contact me](mailto:jess.yan.effort@gmail.com)
 * [Follow me](https://github.com/JessYanCoding)
 * [Star me](https://github.com/JessYanCoding/MVPArms)
 * [See me](https://github.com/JessYanCoding/MVPArms/wiki)
 * [模版请保持更新](https://github.com/JessYanCoding/MVPArmsTemplate)
 * ================================================
 */
@Module
abstract class SettingsModule {

    @Binds
    internal abstract fun bindSettingsModel(model: SettingsModel): SettingsContract.Model
}